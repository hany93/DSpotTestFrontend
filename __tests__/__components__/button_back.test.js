import '@testing-library/jest-dom'
import ButtonBack from '@components/button/button_back'
import { render, screen } from '@testing-library/react'

describe('Render ButtonBack Component', () => {
	beforeEach(() => {
		render(<ButtonBack />)
	})

	it('Test role link', () => {
		expect(screen.getByRole('link').getAttribute('href')).toBe('/')
	})

	it('Test role img', () => {
		expect(screen.getByRole('img')).toBeInTheDocument()
	})

	it('Test className profiles_details__back', () => {
		expect(
			document.querySelector('div .profiles_details__back')
		).toBeInTheDocument()
	})

	it('Test Snapshot', () => {
		var { container } = render(<ButtonBack />)
		expect(container).toMatchSnapshot()
	})
})
