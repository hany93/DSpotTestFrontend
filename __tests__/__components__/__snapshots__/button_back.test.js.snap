// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ButtonBack Component Test Snapshot 1`] = `
<div>
  <div>
    <a
      href="/"
    >
      <div
        class="profiles_details__back"
      >
        <img
          alt="Photo"
          class=""
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`Render ButtonBack Component Test role link and className profiles_details__back 1`] = `
<div>
  <div>
    <a
      href="/"
    >
      <div
        class="profiles_details__back"
      >
        <img
          alt="Photo"
          class=""
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
    </a>
  </div>
</div>
`;
